Improve `opensearch.pluginzip` docs. (#153)
opensearch-project/opensearch-plugins: BUILDING.md

### opensearch.pluginzip

> `opensearch.pluginzip` is designed to facilitate OpenSearch plugin zips to be available in maven repo which can then be fetched for dependency using maven coordinates. This plugin identifies the OpenSearch plugin zip file, the ouput of `bundlePlugin` task and publishes to local maven repo with standard maven coordinates.

Reviewer: ```suggestion
`opensearch.pluginzip` is designed to facilitate OpenSearch plugin zips to be available in maven repo which can then be fetched for dependency using Apache Maven repositories. This plugin identifies the OpenSearch plugin ZIP file, the output of `bundlePlugin` task and publishes to local Apache Maven repository.
```

[Plugin Code](https://github.com/opensearch-project/OpenSearch/tree/main/buildSrc/src/main/java/org/opensearch/gradle/pluginzip), [Plugin Tests](https://github.com/opensearch-project/OpenSearch/tree/main/buildSrc/src/test/java/org/opensearch/gradle/pluginzip), [Plugin META-INF](https://github.com/opensearch-project/OpenSearch/blob/main/buildSrc/src/main/resources/META-INF/gradle-plugins/opensearch.pluginzip.properties)

#### Plugin Design

* `opensearch.pluginzip` is java based code published into build-tools.
* The maven coordinates groupID is fixed as `org.openserach.plugin`, `version` and `artifcatID` will be inferred from gradle project properties.

> * User can also pass custom POM extensions, that will add desired properties to maven POM file generated during runtime.

Reviewer: ```suggestion
* User can also pass custom POM extensions, that will add desired properties to Apache Maven POM file generated during runtime.
```

> * Once the plugin is added to the `build.gradle` as `apply plugin: 'opensearch.pluginzip'`, this will add a new custom publish task `publishPluginZipPublicationToZipStagingRepository`, the purpose of this task is to publish the zip distribution to the maven local (file system).

Reviewer: ```suggestion
* Once the plugin is added to the `build.gradle` as `apply plugin: 'opensearch.pluginzip'`, this will add a new custom publish task `publishPluginZipPublicationToZipStagingRepository`, the purpose of this task is to publish the ZIP distribution to the Apache Maven local repository (file system).
```

> * The maven local artifacts will then be published to maven central/nexus using [CI workflows](https://github.com/opensearch-project/opensearch-build/tree/main/jenkins).

Reviewer: ```suggestion
* The Apache Maven local artifacts could then be published to  Apache Maven central/nexus using [CI workflows](https://github.com/opensearch-project/opensearch-build/tree/main/jenkins).
```

> * The plugin will not add generated jars by tasks `sourcesJar` and `javadocJar`, this is done in purpose to exclude `jars` for `zip` publications.

Reviewer: ```suggestion
* The plugin will not publish generated JARs (`sourcesJar`, `javadocJar`), this is done on purpose to separate `jars` and `zip` publications.
```

3. Add `apply plugin: 'opensearch.pluginzip'` to the build.gradle file.
Once added, this should list the new task `publishPluginZipPublicationToZipStagingRepository`.

4. Run the task `publishPluginZipPublicationToZipStagingRepository` (add it to managed build script `build.sh`)  
```./gradlew publishPluginZipPublicationToZipStagingRepository -Dopensearch.version=$VERSION -Dbuild.snapshot=$SNAPSHOT -Dbuild.version_qualifier=$QUALIFIER```

>     Note: The gradle assemble task `./gradlew assemble` should be called first before calling `publishPluginZipPublicationToZipStagingRepository`, as zip file need to be generated before publishing.

Reviewer 2: Not sure if plugins support `./gradle assemble`, i know few plugins use `./gradle build`[1]

[1] https://github.com/opensearch-project/anomaly-detection/blob/main/DEVELOPER_GUIDE.md#building-from-the-command-line

Author: Thanks for the inputs @saratvemulapalli and @ylwu-amzn, the idea to mention the zip has be available (either by calling `assemble` or `build`) before publishing, let me rephrase this way and let me know this good to go with.

Note: The gradle task responsible to generate the distribution zip should be called first before executing `publishPluginZipPublicationToZipStagingRepository`, as the zip file needs to be generated before publishing.
